common/src/test_utils.rs

```diff
@@ -56,3 +62,808 @@ macro_rules! assert_noop_msg {
 pub fn init_logger() {
     let _ = env_logger::builder().is_test(true).try_init();
 }
+
+#[derive(Error, Debug, Ord, PartialOrd, Eq, PartialEq)]
+pub enum ApproxEqError {
+    #[error("Expected absolute tolerance to be non-negative, got {0:?}")]
+    NegativeAbsoluteTolerance(Fixed),
+    #[error("Expected percentage to be in interval [0, 1), got {0:?}")]
+    IncorrectRelativePercentage(Fixed),
+}
+
+#[inline]
+fn are_approx_eq_abs_unchecked(left: Fixed, right: Fixed, tolerance: Fixed) -> bool {
+    left <= right.saturating_add(tolerance) && right <= left.saturating_add(tolerance)
+}
+
+/// Calculate if two values are approximately equal up to some absolute tolerance (constant value).
+///
+/// **NOTE**: It is preferred to utilize to exact equalities even in tests. Fixed point arithmetic
+/// allows predictable behaviour of inner arithmetics, so it should be considered everywhere where
+/// possible. Use approximate equalities only when you know what you're doing and such inaccurate
+/// behavior is expected.
+pub fn are_approx_eq_abs(
+    left: Fixed,
+    right: Fixed,
+    tolerance: Fixed,
+) -> Result<bool, ApproxEqError> {
+    if tolerance >= Fixed::ZERO {
+        Ok(are_approx_eq_abs_unchecked(left, right, tolerance))
+    } else {
+        Err(ApproxEqError::NegativeAbsoluteTolerance(tolerance))
```

Comment:
This line is not covered. It would be nice to add a trivial test [like here](https://github.com/sora-xor/sora2-network/pull/869/files#diff-702d59a8c1fa0fa009bc2168d0ad50226dd4a6ce0e37754a86baee08185322ccR857-R869).